From b6fba62cc2ca2a458ba46a0f798edfec3d300cf1 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 7 Jul 2020 14:23:40 +0100
Subject: [PATCH] media: bcm2835-unicam: Reinstate V4L2_CAP_READWRITE in the
 caps

v4l2-compliance throws a failure if the device doesn't advertise
V4L2_CAP_READWRITE but allows read or write operations.
We do support read, so reinstate the flag.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/platform/bcm2835/bcm2835-unicam.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/bcm2835/bcm2835-unicam.c b/drivers/media/platform/bcm2835/bcm2835-unicam.c
index 0282c5fb7a5c..4118bf7ca042 100644
--- a/drivers/media/platform/bcm2835/bcm2835-unicam.c
+++ b/drivers/media/platform/bcm2835/bcm2835-unicam.c
@@ -2420,8 +2420,8 @@ static int register_node(struct unicam_device *unicam, struct unicam_node *node,
 	vdev->queue = q;
 	vdev->lock = &node->lock;
 	vdev->device_caps = (pad_id == IMAGE_PAD) ?
-			    (V4L2_CAP_VIDEO_CAPTURE | V4L2_CAP_STREAMING) :
-			    (V4L2_CAP_META_CAPTURE | V4L2_CAP_STREAMING);
+				V4L2_CAP_VIDEO_CAPTURE : V4L2_CAP_META_CAPTURE;
+	vdev->device_caps |= V4L2_CAP_READWRITE | V4L2_CAP_STREAMING;
 
 	/* Define the device names */
 	snprintf(vdev->name, sizeof(vdev->name), "%s-%s", UNICAM_MODULE_NAME,
-- 
2.35.5

